app/helpers/report_helper.rb

```diff
@@ -3,15 +3,24 @@
 #   the COPYRIGHT file.
 
 module ReportHelper
+  def get_reported_guid(id, type)
+    if type == "post"
+      Post.where(id: id).first.author.guid
+    elsif type == "comment"
+      Comment.where(id: id).first.author.guid
+    end
+  end
+
   def report_content(id, type)
-    if type == 'post' && !(post = Post.find_by_id(id)).nil?
-      raw t('report.post_label', title: link_to(post_page_title(post), post_path(id)))
-    elsif type == 'comment' && !(comment = Comment.find_by_id(id)).nil?
+    if type == "post" && !(post = Post.find_by_id(id)).nil?
+      raw t("report.post_label", title: link_to(post_page_title(post), post_path(id)))
+    elsif type == "comment" && !(comment = Comment.find_by_id(id)).nil?
       # comment_message is not html_safe. To prevent
       # cross-site-scripting we have to escape html
-      raw t('report.comment_label', data: link_to(h(comment_message(comment)), post_path(comment.post.id, anchor: comment.guid)))
+      raw t("report.comment_label", data: link_to(
+        h(comment_message(comment)), post_path(comment.post.id, anchor: comment.guid)))
     else
-      raw t('report.not_found')
+      raw t("report.not_found")
     end
```

Comment:
Also while touching this I feel like we can clean it up with a method or association on the Report model or two so it becomes

```rb
case report.item
when Post
  # ...
when Comment
  # ...
else
  # not found
end